libs/sdk-ui-dashboard/src/plugins/customizer.ts

```diff
@@ -0,0 +1,187 @@
+// (C) 2021 GoodData Corporation
+import React from "react";
+import { InsightComponentProvider } from "../presentation";
+import {
+    DashboardDispatch,
+    DashboardEventHandler,
+    DashboardEventHandlerFn,
+    DashboardEvents,
+    DashboardEventType,
+    DashboardState,
+    ICustomDashboardEvent,
+} from "../model";
+
+/**
+ * @alpha
+ */
+export interface IDashboardInsightCustomizer {
+    /**
+     * A convenience method that will register a specific React component to use for rendering
+     * any insight that is tagged with the provided `tag`. If plugins register multiple providers
+     * for the same tag, then the provider will be picked using 'last-win' strategy.
+     *
+     * @param tag - tag to look for on the insight
+     * @param component - component to use if the tag is found
+     * @returns self, for call chaining sakes
+     */
+    withTag(tag: string, component: React.ComponentType): IDashboardInsightCustomizer;
+
+    /**
+     * Register a provider for React components to render insights. A provider takes the insight and
+     * widget that it is part of as input and is expected to return a React component that should be
+     * used to render that insight.
+     *
+     * If the provider returns `undefined` then:
+     *
+     * -  if there are other providers registered, they will be called to see if they can provide
+     *    a component to render the insight
+     * -  if there are no other providers registered, the default, built-in component will be used.
+     *
+     * You may register multiple providers. They will be evaluated in the order you register them.
+     *
+     * @remarks see the {@link IDashboardInsightCustomizer.withTag} convenience method to register components for insights
+     *  with particular tags.
+     * @param provider - provider to register
+     * @returns self, for call chaining sakes
+     */
+    withCustomProvider(provider: InsightComponentProvider): IDashboardInsightCustomizer;
+
+    /**
+     * Register a factory for insight decorator providers. Decorators are a way to add customizations or embellishments on top
+     * of an existing component. Decorators are more complex to write because they need to work with the component
+     * they should decorate and add 'something' on top of that component.
+     *
+     * This is best illustrated on an example:
```

Comment:
Nitpick: please use the [`@example`](https://tsdoc.org/pages/tags/example/) tag, it will make the intellisense in IDEs better formatted.

Reply:
changed